libafl_frida/src/alloc.rs

```diff
@@ -562,14 +578,70 @@ impl Allocator {
     }
 
     /// Unpoison all the memory that is currently mapped with read permissions.
+    #[cfg(target_vendor = "apple")]
+    pub fn unpoison_all_existing_memory(&mut self) {
+        let task = unsafe { mach_task_self() };
+        let mut address: mach_vm_address_t = 0;
+        let mut size: mach_vm_size_t = 0;
+        let mut depth: natural_t = 0;
+
+        loop {
+            let mut kr;
+            let mut info_count: mach_msg_type_number_t = VM_REGION_SUBMAP_INFO_COUNT_64;
+            let mut info = vm_region_submap_info_64::default();
+            loop {
+                kr = unsafe {
+                    mach_vm_region_recurse(
+                        task,
+                        addr_of_mut!(address),
+                        addr_of_mut!(size),
+                        addr_of_mut!(depth),
+                        addr_of_mut!(info) as vm_region_recurse_info_t,
+                        addr_of_mut!(info_count),
+                    )
+                };
+
+                if kr != KERN_SUCCESS {
+                    break;
+                }
+
+                if info.is_submap != 0 {
+                    depth += 1;
+                    continue;
+                } else {
+                    break;
+                }
+            }
+
+            if kr != KERN_SUCCESS {
+                break;
+            }
+
+            let start = address as usize;
+            let end = (address + size) as usize;
+
+            if info.protection & VM_PROT_READ == VM_PROT_READ {
+                //if its at least readable
+                if self.shadow_offset <= start && end <= self.current_mapping_addr {
```

Comment:
shouldn't the right hand side of the second condition be `self.base_mapping_addr`?

Reply:
oh wait. no never mind.